src/Chapter2.hs

```diff
@@ -741,7 +757,13 @@ the list with only those lists that contain a passed element.
 
 🕯 HINT: Use the 'elem' function to check whether an element belongs to a list
 -}
-contains = error "contains: Not implemented!"
+contains ::  Int -> [[Int]] -> [[Int]]
+contains _ [] = []
+contains i list = go i list
+  where
+    go :: Int -> [[Int]] -> [[Int]]
+    go _ [] = [] 
+    go item (x:xs) = if elem item x then x : go item xs else go item xs
```

Comment:
That's a correct implementation 👍🏻  And you correctly use `elem` to check if the element is in the list.

Here you can use the standard function `filter` to automatically filter the list by the given predicate. So you won't need to implement manual recursion.

Reply:
I did it with filter to see how it looks like, I see your point